iaso/institutions/entity/matching/types.py

```diff
@@ -0,0 +1,50 @@
+from collections import defaultdict
+from enum import Enum, auto
+
+
+class Entity(Enum):
+    IRRELEVANT = auto()
+    ORGANISATION = auto()
+    LOCATION = auto()
+
+
+async def fetch_wikidata_entity_types(client, wikidata_entities):
+    query = (
+        """SELECT DISTINCT ?entity ?superclass WHERE {
+      VALUES ?entity { """
+        + " ".join(f"wd:{qid}" for qid in wikidata_entities)
+        + """ }
+
+      ?entity wdt:P31 ?class.
+      ?class wdt:P279* ?superclass.
+    }"""
+    )
+
+    response = await client.query(query)
+
+    entity_superclasses = defaultdict(set)
+
+    for result in response["results"]["bindings"]:
+        entity_superclasses[result["entity"]["value"][31:]].add(
+            result["superclass"]["value"][31:]
+        )
+
+    entity_types = dict()
+
+    for entity, superclasses in entity_superclasses.items():
+        is_organisation = "Q43229" in superclasses
```

Comment:
Wikidata values?
Let's have a chat on this one in the next office day

Reply:
As I have discussed with you before, the institution extraction and linking uses the Wikidata knowledge graph and ontology to find relevant entries, make sure they are not in completely wrong categories, and find additional information about them.

Reply:
In the long run, we should maybe think about externalising those values to make them configurable